elfpy/data/convert_data.py

```diff
@@ -0,0 +1,389 @@
+"""Script to format on-chain hyperdrive pool, config, and transaction data post-processing."""
+from __future__ import annotations
+
+import logging
+import time
+from typing import Any
+
+from eth_typing import BlockNumber
+from fixedpointmath import FixedPoint
+from hexbytes import HexBytes
+from web3 import Web3
+from web3.contract.contract import Contract
+from web3.types import BlockData
+
+from elfpy import eth, hyperdrive_interface
+from elfpy.data import db_schema
+
+# pylint: disable=too-many-arguments
+
+# TODO fix too many branches by splitting out various things into functions
+# pylint: disable=too-many-branches
+
+RETRY_COUNT = 10
+
+
+def _convert_scaled_value(input_val: int | None) -> float | None:
+    """
+    Given a scaled value int, converts it to a float, while supporting Nones
+
+    Arguments
+    ----------
+    input_val: int | None
+        The scaled integer value to unscale and convert to float
+
+    Returns
+    -------
+    float | None
+        The unscaled floating point value
+
+    Note
+    ----
+    We cast to FixedPoint, then to floats to keep noise to a minimum.
+    There is no loss of precision when going from Fixedpoint to float.
+    Once this is fed into postgres, postgres will use the fixed-precision Numeric type.
+    """
+    if input_val is not None:
+        return float(FixedPoint(scaled_value=input_val))
+    return None
+
+
+# TODO move this function to hyperdrive_interface and return a dictionary
+def fetch_transactions_for_block(
```

Comment:
ok if this is for a particular contract, let's indicate that:  `fetch_contract_transactions_for_block`